content/sse/index.textile

```diff
@@ -0,0 +1,119 @@
+
+Due to this feature being in beta, it is currently subject to other restrictions. For example, at the moment you can not resume from part-way through a backlog of messages that was being sent as part of a previous resume; you can only resume from a message (so you can't resume unless you've received at least one message on the connection); and you can only resume from the id of a message that was sent in the last two minutes (so it will not work if the last message sent on the connection was more than two minutes ago). We expect to fix all of these by the time this feature graduates out of beta.
+
+h1(#api). API routes
+
+h3(#sse). Server-sent events
+
+h6. GET realtime.ably.io/sse
+
+Start a streaming HTTP request that conforms to the "Server-Sent Events":https://www.w3.org/TR/eventsource/ spec, for ease of consuming with an SSE library.
+
+The @/event-stream@ endpoint will give an SSE response if the @Accept@ header is set to @text/event-stream@. The @/sse@ endpoint is also provided as an easier way of forcing an SSE response.
+
+h5. Request parameters
+
+Identical to the request parameters for a plain event stream above.
+
+h5(#envelope-format). Envelope format
+
+See for a plain event stream above, except instead of a JSON object with @id@, @event@, @data@ members, you get an SSE event.
+
+Keepalive packets are sent as sse comments (@:keepalive@).
+
+h5. Example
+
+```[sh]
+curl "https://rest.ably.io/sse?key={{API KEY}}&channel=example&v=1.1"
+⏎
+id: cbfKayrzgAXDWM:1556806691343-0
+event: message
+data: {"id":"YqigX7VFsR:0:0","name":"foo","timestamp":1556806691341,"encoding":"json","channel":"channel","data":"{\"foo\":1}"}
+⏎
+:keepalive
+⏎
+event: error
+data:{"message":"Access token expired. (See https://help.ably.io/error/40142 for help.)","code":40142,"statusCode":401,"href":"https://help.ably.io/error/40142"}
+```
+
+h3(#event-stream). Plain event stream
+
+h6. GET realtime.ably.io/event-stream
+
+Start a streaming HTTP request.
+
+h5. Request parameters
+
+- channels := (mandatory) one or more channel names, separated by commas (or the @separator@ if specified). Non-url-safe characters should be URL-encoded (for example, @?channels=foo%3Fbar@ will subscribe to the channel @foo?bar@). Alias: @channel@.
+- separator := (optional) a separator, to enable easy subscriptions to channels with commas in their name. For example, @?separator=|&channel=fo,o|ba,r@ will subcribe to the two channels @fo,o@ and @ba,r@.
+- v := (mandatory) the version of the api you are requesting. The current (and only) version of the API is 1.1, so only @v=1.1@ is accepted.
+- key := (optional) an Ably API key to use, if using basic auth.
+- accessToken := (optional) an Ably auth token to use, if using token auth.
+- lastEvent := (optional) an @id@ to resume from, see "connection state recovery":#recovery below.
```

Comment:
@SimonWoolf do we additionally support `rewind` now?

Reply:
Is `lastEvent` supported for SSE, or more accurately, should we tell people it's not needed except when starting a new SSE connection that resumes from a previous connection?

Reply:
This recovery link does not exist, and refers to a link below, which does not exist (it's above).

Reply:
> do we additionally support rewind now?

Yes (currently just with a number though)

> Is lastEvent supported for SSE, or more accurately, should we tell people it's not needed except when starting a new SSE connection that resumes from a previous connection?

sgtm (if supplied it takes precedence over the SSE `headers['last-event-id']`)

Reply:
How is `rewind` to be defined @SimonWoolf ?

Reply:
`- rewind := (optional integer) if specified, will send a backlog of this many messages once the connection opens. For example, @rewind=1@ will give you the most recent message sent on the channel. This is best-effort — only messages from within the last two minutes will be available, and only if the channel has been continuously active since the message was sent; it is not a replacement for the "history API":/rest/history. It only has an effect for new connections; when "resuming a previous connection":#connection-state-recovery, it is ignored in favour of sending you the messages you missed since you were last connected`